### DISCUSSION TITLE
Fleets and scanners?

### DISCUSSION THREAD

**Discussion by junetried** [endless-sky/endless-sky#8495] [Ideas]
Unless I'm mistaken, currently there is no point to equipping any scanners to ships that are not your flagship. I feel like there should be some practical use for scanners equipped on your fleet.

My first idea would be to add a scan command to make an escort to attempt to scan the selected ship. That could also give drones like the Republic surveillance drone and the Tern a new niche as being safe scanners, which is already what they seem to be normally used for in-lore anyway.

Similarly, the tactical scanner information could be visible if any ship in your fleet equipped with one is close enough to the targeted ship to see it.

I'm not sure how this would work for asteroid scanners, though. Maybe a ship with an asteroid scanner chases around asteroids and makes them visible to your flagship? Maybe asteroids being visible is just a passive effect around the ship equipped with the scanner? Or maybe asteroids are only visible to the ship with the scanner, but you can command it to mine the asteroid?

**Comment by bene-dictator** [comment]
If there is ever substantial AI inaccuracy / skill statistics (there is some sort of AI `confusion` variable but it's not very impactful), Tactical Scanners might decrease the AI's inaccuracy and make them more confident or proficient in combat instead of being dead outfit space. For example, with these outfits they'll try to stay out of range of enemy ships' weaponry, strafe and keep back, etc., but without it they'll be less aware of their weapons' effective range and stray too close to enemies' effective range.

In a similar vein, there have been EV-style radar proof of concepts before where if you have no radar upgrades, you'll see very basic information on your radar. There would be outfits to make them more accurate (radar refresh times) and give you better information (ship type, IFF, etc.) Combined with the last suggestion, the AI's proficiency in combat might be improved with those radar outfits. Or, they might relay information towards you, giving your radar a bigger view to where enemy ships lie.

There are also plans to improve asteroid mining. In the past, there was a testing PR where escort ships with the Asteroid Scanner installed would automatically start mining. Unfortunately I don't know if that is being considered right now.

**Reply by junetried** [reply]
Changing AI behavior for having tactical scanning isn't an idea I'm totally against, but it doesn't help outfit and cargo scanners.

**Comment by GPaso** [comment]
I was wondering about that too and I think that there is no need of a new key, it should just be default that when you scan, all your ships that have a scanner will try to scan the selected ship, unless you could implement some sort of previously and currently selected so that the previously selected ship (your escort ship, if it has a scanner) will try to scan the currently selected ship.

Also like when you have multiple scanners in your ship it enhances the scanning power, it should apply when you have more ships with scanners.

As per the AI behaviour with the tactical scanning could be a great add but needs more developement.
